src/graph/immutable_graph.cc

```diff
@@ -549,7 +550,8 @@ COO::EdgeArray COO::Edges(const std::string &order) const {
   return EdgeArray{src_, dst_, rst_eid};
 }
 
-Subgraph COO::EdgeSubgraph(IdArray eids) const {
+Subgraph COO::EdgeSubgraph(IdArray eids, bool preserve_nodes) const {
+  assert(!preserve_nodes); // TODO(zihao): handle the case of preserve_nodes == true
```

Comment:
Could you help implement this? It shouldn't be too hard. Just create a COO with the given src,dst id and the same number of nodes.